dwave/optimization/symbols.pyx

```diff
@@ -762,11 +686,11 @@ cdef class BinaryVariable(ArraySymbol):
 
         self.initialize_arraynode(model, self.ptr)
 
-    @staticmethod
-    def _from_symbol(Symbol symbol):
+    @classmethod
+    def _from_symbol(cls, Symbol symbol):
         cdef cppBinaryNode* ptr = dynamic_cast_ptr[cppBinaryNode](symbol.node_ptr)
         if not ptr:
-            raise TypeError("given symbol cannot be used to construct a BinaryVariable")
+            raise TypeError(f"given symbol cannot construct a {cls.__name__}")
```

Comment:
I see this change in the error string happens only here and not consistently through the file. You left the original error string elsewhere

Reply:
You're right. It's an internal error so I doubt it matters much. But sure, I can change it to be consistent elsewhere.